src/ed25519/mod.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    constants::ED25519_TAG,
+    crypto_tools::{message_digest::MessageDigest, rng},
+    sdk::api::{BytesVec, TofnFatal, TofnResult},
+};
+use ed25519_dalek::{Signature, Signer, SigningKey, VerifyingKey, PUBLIC_KEY_LENGTH};
+
+#[derive(Debug)]
+pub struct KeyPair(SigningKey);
+
+impl KeyPair {
+    pub fn encoded_verifying_key(&self) -> [u8; PUBLIC_KEY_LENGTH] {
+        *self.0.verifying_key().as_bytes()
+    }
+}
+
+pub fn keygen(
+    secret_recovery_key: &rng::SecretRecoveryKey,
+    session_nonce: &[u8],
+) -> TofnResult<KeyPair> {
+    let mut rng = rng::rng_seed_ecdsa_signing_key(
```

Comment:
It might be worth renaming `rng_seed_ecdsa_signing_key`, I don't think there's anything specfiic in it for ECDSA

Reply:
Makes sense, perhaps `rng_seed_signing_key`?